collector/mongod/wiredtiger.go

```diff
@@ -0,0 +1,403 @@
+	Read	*WTConcurrentTransactionsTypeStats	`bson:"write"`
+}
+
+func (stats *WTConcurrentTransactionsStats) Export(ch chan<- prometheus.Metric) {
+	wtConcurrentTransactionsOut.WithLabelValues("read").Set(stats.Read.Out)
+	wtConcurrentTransactionsOut.WithLabelValues("write").Set(stats.Write.Out)
+	wtConcurrentTransactionsAvailable.WithLabelValues("read").Set(stats.Read.Available)
+	wtConcurrentTransactionsAvailable.WithLabelValues("write").Set(stats.Write.Available)
+	wtConcurrentTransactionsTotalTickets.WithLabelValues("read").Set(stats.Read.TotalTickets)
+	wtConcurrentTransactionsTotalTickets.WithLabelValues("write").Set(stats.Write.TotalTickets)
+}
+
+func (stats *WTConcurrentTransactionsStats) Describe(ch chan<- *prometheus.Desc) {
+	wtConcurrentTransactionsOut.Describe(ch)
+	wtConcurrentTransactionsAvailable.Describe(ch)
+	wtConcurrentTransactionsTotalTickets.Describe(ch)
+}
+
+// WiredTiger stats
+type WiredTigerStats struct {
+	Async			*WTAsyncStats			`bson:"async"`
+	BlockManager		*WTBlockManagerStats		`bson:"block-manager"`
+	Cache			*WTCacheStats			`bson:"cache"`
+	Connection		*WTConnectionStats		`bson:"connection"`
+	Cursor			*WTCursorStats			`bson:"cursor"`
+	Log			*WTLogStats			`bson:"log"`
+	Session			*WTSessionStats			`bson:"session"`
+	Transaction		*WTTransactionStats		`bson:"transaction"`
+	ConcurrentTransactions	*WTConcurrentTransactionsStats	`bson:"concurrentTransactions"`
+}
+
+func (stats *WiredTigerStats) Describe(ch chan<- *prometheus.Desc) {
+	if stats.Async != nil {
+		stats.Async.Describe(ch)
+	}
+	if stats.BlockManager != nil {
+		stats.BlockManager.Describe(ch)
+	}
+	if stats.Cache != nil {
+		stats.Cache.Describe(ch)
+	}
+	if stats.Transaction != nil {
+		stats.Transaction.Describe(ch)
+	}
+	if stats.ConcurrentTransactions != nil {
+		stats.ConcurrentTransactions.Describe(ch)
+	}
+
+	wtBlockManagerBlocksTotal.Describe(ch)
+	wtBlockManagerBytesTotal.Describe(ch)
+
+	wtCachePagesTotal.Describe(ch)
+	wtCacheBytesTotal.Describe(ch)
+	wtCacheEvictedTotal.Describe(ch)
+	wtCacheCurPages.Describe(ch)
+	wtCacheBytesCached.Describe(ch)
+	wtCacheBytesMax.Describe(ch)
+	wtCachePercentOverhead.Describe(ch)
+
+	wtTransactionsTotal.Describe(ch)
+	wtTransactionsTotalCheckpointMs.Describe(ch)
+	wtTransactionsCheckpointsRunning.Describe(ch)
+
+	wtConcurrentTransactionsOut.Describe(ch)
+	wtConcurrentTransactionsAvailable.Describe(ch)
+	wtConcurrentTransactionsTotalTickets.Describe(ch)
+
+	wtAsyncWorkQueueLength.Describe(ch)
+	wtAsyncMaxWorkQueueLength.Describe(ch)
+}
+
+func (stats *WiredTigerStats) Export(ch chan<- prometheus.Metric) {
+	if stats.Async != nil {
+		stats.Async.Export(ch)
+	}
+	if stats.BlockManager != nil {
+		stats.BlockManager.Export(ch)
+	}
+	if stats.Cache != nil {
+		stats.Cache.Export(ch)
+	}
+	if stats.Transaction != nil {
+		stats.Transaction.Export(ch)
+	}
+	if stats.ConcurrentTransactions != nil {
+		stats.ConcurrentTransactions.Export(ch)
+	}
+
+	wtBlockManagerBlocksTotal.Collect(ch)
```

Comment:
I'm not sure on all accounts. I started small and added until I felt like I could "see enough" of what WT was doing. I would say this is the bare minimum we might need, and more can be added.